### PR TITLE
[Enabler][Zos Job Submit]Update undocumented argument and import exception

### PR DESCRIPTION
**Pull request by AndreMarcel99**
##### SUMMARY
Update temp_path and how to import exceptions.

Fiexes #997 #998 #999 

##### ISSUE TYPE
- Sanity Issue Pull Request

##### ADDITIONAL INFORMATION
Remove tmp path from options and only work with source as name of tmp file, because it was undocumented argument for ansible and could be any name, also only in case is local make the tmp file, so didn't expect a dataset or file with the same name on the system. Also make a implementation of imports of exceptions by using new implementation to avoid losing context when an import fails.

Pipeline Without Zos_copy
Captura de pantalla 2023-12-12 a la(s) 4 08 41 p m
Captura de pantalla 2023-12-12 a la(s) 4 09 21 p m

Pipeline with zos_copy

### CODE REVIEW

File: plugins/modules/zos_job_submit.py [hunk 1]
```diff
@@ -902,7 +899,7 @@ def run_module():
     from_encoding = parsed_args.get("from_encoding")
     to_encoding = parsed_args.get("to_encoding")
     # temporary file names for copied files when user sets location to LOCAL
-    temp_file = parsed_args.get("temp_file")
+    temp_file = parsed_args.get("src")
```
Comment (fernandofloresg) [inline plugins/modules/zos_job_submit.py:902]: I would suggest doing this change instead : 

```suggestion
    if location == "LOCAL":
        temp_file = parsed_args.get("src")
```

That way `temp_file` is indeed a temporary file on the remote and we don't need to worry on the `location` conditions in the future, if `temp_file` is None we know we don't have to remove anything.

File: plugins/modules/zos_job_submit.py [hunk 2]
```diff
@@ -1039,7 +1036,7 @@ def run_module():
         module.exit_json(**result)
 
     finally:
-        if temp_file:
+        if location != "DATA_SET" and location != "USS":
```
Comment (fernandofloresg) [inline plugins/modules/zos_job_submit.py:1039]:
```suggestion
        if temp_file:
```

I suggest keeping it this way bc that will help us avoid having to check for state and just remove it if it exists.

**Review by fernandofloresg — CHANGES_REQUESTED** [review]
Requested changes to avoid tech debt in the future and also add a changelog please.

File: plugins/modules/zos_job_submit.py (hunk 2 above)
Comment (AndreMarcel99) [reply plugins/modules/zos_job_submit.py:1039]: It generates a error if the local file is named as a dataset (USER.LOCAL) then the file /user.local will fail that verification

Comment (fernandofloresg) [reply plugins/modules/zos_job_submit.py:1039]: I'm sorry I don't get how that relates to this code suggestion, could you explain?

**Comment by AndreMarcel99** [comment]
Pipeline for all except zos_copy
Captura de pantalla 2023-12-14 a la(s) 10 22 01 a m

Captura de pantalla 2023-12-14 a la(s) 10 21 47 a m

**Comment by AndreMarcel99** [comment]
Pipeline zos_copy
Captura de pantalla 2023-12-14 a la(s) 2 34 44 p m
Captura de pantalla 2023-12-14 a la(s) 2 35 19 p m

**Comment by fernandofloresg** [comment]
I made some extra changes and I'm waiting for the pipeline to finish.

**Comment by fernandofloresg** [comment]
Screenshot 2023-12-15 at 12 15 28 PM

**Comment by fernandofloresg** [comment]
To add in case anyone wonders why `iconv` section is because when `zos_copy` module was integrated into `zos_job_submit` encoding option was not used, this way we can leverage `zos_copy` encoding capabilities and standardize copying files into remote.

**Review by fernandofloresg — APPROVED** [review]
Looks good to me now

**Review by richp405 — APPROVED** [review]
Looks good!

**Review by rexemin — APPROVED** [review]

**Pull request merged**